In [1]:
import pandas as pd

In [2]:
titanic_df = pd.read_csv("titanic_data.csv")
titanic_df.head()

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S


In [3]:
titanic_df.shape

(891, 12)

## Conditional Based Filtering

In [3]:
mask = titanic_df.Fare > 100

In [7]:
fare_100 = titanic_df[mask]

In [10]:
fare_100.shape

(53, 12)

Out of 891 passangers , only 53 passangers paid more than 100 $ in fare.

In [11]:
female_fare_100 = fare_100[ fare_100.Sex == "female" ]

In [12]:
female_fare_100.shape

(34, 12)

In [16]:
print(f"Out of 53, {round((34/53) * 100, 2)} % are female passengers")
## Why nearly 65% of highest paying customer are female in 1912 ?

Out of 53, 64.15 % are female passengers


In [17]:
fare_100_male = titanic_df[
    (titanic_df.Fare > 100) & (titanic_df.Sex == "male")
]

In [20]:
fare_100_male.shape

(19, 12)

<< Write Description or result >>

In [21]:
# What is Survival Rate of Higher-Class passengers ?

hight_class_survival = titanic_df[
    (titanic_df.Pclass == 1) & (titanic_df.Survived == 1)
]

In [29]:
high_class = titanic_df[(titanic_df.Pclass == 1)]
high_class.shape

(216, 12)

In [23]:
hight_class_survival.shape

(136, 12)

## Filtering using existing methods

In [9]:
# isna()
null_cabin = titanic_df[titanic_df.Cabin.isna()]

In [10]:
null_cabin.head()

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S
5,6,0,3,"Moran, Mr. James",male,NaN,0,0,330877,8.4583,NaN,Q
7,8,0,3,"Palsson, Master. Gosta Leonard",male,2.0,3,1,349909,21.0750,NaN,S


In [11]:
null_cabin.shape

(687, 12)

In [13]:
# Total Null values
titanic_df.Cabin.isna().sum()

np.int64(687)

In [19]:
# notna()
non_empty_cabin = titanic_df[ titanic_df.Cabin.notna()]
non_empty_cabin.shape

(204, 12)

In [30]:
# duplicated data
duplicate_ticket = titanic_df[titanic_df.duplicated(subset=["Ticket"], keep = "first")] 
duplicate_ticket.head()

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
24,25,0,3,"Palsson, Miss. Torborg Danira",female,8.0,3,1,349909,21.075,NaN,S
71,72,0,3,"Goodwin, Miss. Lillian Amy",female,16.0,5,2,CA 2144,46.900,NaN,S
88,89,1,1,"Fortune, Miss. Mabel Helen",female,23.0,3,2,19950,263.000,C23 C25 C27,S
117,118,0,2,"Turpin, Mr. William John Robert",male,29.0,1,0,11668,21.000,NaN,S
119,120,0,3,"Andersson, Miss. Ellis Anna Maria",female,2.0,4,2,347082,31.275,NaN,S


In [28]:
titanic_df[titanic_df.Ticket == "349909"]

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
7,8,0,3,"Palsson, Master. Gosta Leonard",male,2.0,3,1,349909,21.075,NaN,S
24,25,0,3,"Palsson, Miss. Torborg Danira",female,8.0,3,1,349909,21.075,NaN,S
374,375,0,3,"Palsson, Miss. Stina Viola",female,3.0,3,1,349909,21.075,NaN,S
567,568,0,3,"Palsson, Mrs. Nils (Alma Cornelia Berglund)",female,29.0,0,4,349909,21.075,NaN,S


In [33]:
# Duplicated row
titanic_df[titanic_df.duplicated()]

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked


In [37]:
# isin
titanic_df[titanic_df.Ticket.isin(["349909"])]

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
7,8,0,3,"Palsson, Master. Gosta Leonard",male,2.0,3,1,349909,21.075,NaN,S
24,25,0,3,"Palsson, Miss. Torborg Danira",female,8.0,3,1,349909,21.075,NaN,S
374,375,0,3,"Palsson, Miss. Stina Viola",female,3.0,3,1,349909,21.075,NaN,S
567,568,0,3,"Palsson, Mrs. Nils (Alma Cornelia Berglund)",female,29.0,0,4,349909,21.075,NaN,S


In [48]:
# where
# mask

titanic_df.Age.where( titanic_df.Age > 0 , other= 1)
# titanic_df.Age.mask( titanic_df.Age < 0 , other= 1)

0      22.0
1      38.0
2      26.0
3      35.0
4      35.0
       ... 
886    27.0
887    19.0
888     1.0
889    26.0
890    32.0
Name: Age, Length: 891, dtype: float64

## Filter using index

Syntax : ```dataframe.loc[<row index> , <column index>]```

In [51]:
titanic_df.loc[:10, ["Name","Age"]]

,Name,Age
0,"Braund, Mr. Owen Harris",22.0
1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",38.0
2,"Heikkinen, Miss. Laina",26.0
3,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",35.0
4,"Allen, Mr. William Henry",35.0
5,"Moran, Mr. James",NaN
6,"McCarthy, Mr. Timothy J",54.0
7,"Palsson, Master. Gosta Leonard",2.0
8,"Johnson, Mrs. Oscar W (Elisabeth Vilhelmina Berg)",27.0
9,"Nasser, Mrs. Nicholas (Adele Achem)",14.0


In [57]:
titanic_df.loc[890, "Age"]

np.float64(32.0)

In [60]:
titanic_df.at[890, "Age"]

np.float64(32.0)